service/src/main/java/org/broadinstitute/listener/relay/inspectors/RequestLogger.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.listener.relay.inspectors;
+
+import com.microsoft.azure.relay.RelayedHttpListenerRequest;
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.TreeMap;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.listener.relay.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequestLogger {
```

Comment:
I converted the HeaderRequestLogger from an `Inspector` to a POJO. The Inspector abstraction was proving to be leaky and not suited to what we need here. Specifically, the ability to inspect the request headers (for cookie + token info), _and_ the response results (status code). 

Rather than stretching that interface and mucking with the chain to create a post-processor abstraction, I opted to make this a POJO and insert a logging statement in the `RelayedHttpRequestProcessor` class. If we find ourselves needing a real chain of command for postprocessing ,we can break out a fancy interface and processing chain similar to what we're doing for requests, but that feels like overkill for now.

Reply:
This makes a lot of sense, feels right to have explicit logging 👍